etcdserver/server.go

```diff
@@ -1082,8 +1082,18 @@ func (s *EtcdServer) applyEntryNormal(e *raftpb.Entry) {
 		id = raftReq.Header.ID
 	}
 
-	ar := s.applyV3.Apply(&raftReq)
+	// set the consistent index of current executing entry
+	s.consistIndex.setConsistentIndex(e.Index)
+	var ar *applyResult
+	if s.w.IsWaiting(id) || !noSideEffect(&raftReq) {
```

Comment:
> !s.w.IsWaiting(id) && noSideEffect(&raftReq)

It is an inverse of the required condition. Do you mean `!(!s.w.IsWaiting(id) && noSideEffect(&raftReq))` ?